### PR TITLE
Allow configuring CASC Reload Token with environment variable

### LINKED ISSUE #1839: Allow setting the CASC Reload Token by environment variable
### What feature do you want to see added?

According to the documentation on [Triggering Configuration Reload](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/configurationReload.md), one can configure a secure token to allow reloading the configs using the `JENKINS_URL/reload-configuration-as-code` endpoint. This is even used on [Jenkins' Helm Chart](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/templates/jenkins-controller-statefulset.yaml#L187).

However, the only way the token can be configured is by setting it as a system property `casc.reload.token`. Java system properties are commonly configured through the command line and appear on the process list, meaning anyone with access to the machine can see them:

```sh
$ pgrep -af -- '-Xmx4G'
1234 /etc/alternatives/java -Xmx4G -Dcasc.reload.token="yourSecretTokenHere" -Djava.awt.headless=true -DJENKINS_HOME=/var/lib/jenkins -jar /usr/lib/jenkins/jenkins.war ...
```

To help with this issue, we could add, in similar fashion to the `CASC_JENKINS_CONFIG` environment variable, a `CASC_RELOAD_TOKEN` environment variable, which would avoid showing the token on the process list and would help when bootstrapping Jenkins while setting it up with Docker or on Kubernetes.

### Upstream changes

_No response_

### PR DESCRIPTION
**Pull request by Angelin01**
This PR adds the following:

* The ability to configure the CASC Reload Token using the `CASC_RELOAD_TOKEN` environment variable.
* Said environment variable takes precedence over the `casc.reload.token` system property.
* Two tests were created to verify the two features above.
* The documentation was updated to reflect these changes. Also, a note was added on the required trailing slash on the reload endpoint and some extra tips on how to hide the token from the process list. The section on reloading with this token was also moved up, since it appears to be the easiest way to perform this operation.

Closes #1839 

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: docs/features/configurationReload.md
Comment (lemeurherve) [inline docs/features/configurationReload.md:16]: How about suggesting to set a local environment variable as a less trickery alternative?

(Thanks for it in any case, I didn't kew it)

Comment (Angelin01) [reply docs/features/configurationReload.md:16]: A local environment variable will still be expanded and show on the process list, as it's expanded by the shell:

```sh
angelo@irgeshud:~ > CASC_RELOAD_TOKEN=someSecretValue
# I used 10.0.0.0 to make sure curl hangs
angelo@irgeshud:~ > curl -X POST http://10.0.0.0:8080/reload-configuration-as-code/?casc-reload-token=$CASC_RELOAD_TOKEN 
# On another shell...
angelo@irgeshud:~ > pgrep -af curl
32701 curl -X POST http://10.0.0.0:8080/reload-configuration-as-code/?casc-reload-token=someSecretValue
```

With the trickery however:

```sh
angelo@irgeshud:~ > curl -X POST -G -d @<(cat <<< "casc-reload-token=$CASC_RELOAD_TOKEN") http://10.0.0.0:8080/reload-configuration-as-code/
# On another shell...
angelo@irgeshud:~ > pgrep -af curl
32940 curl -X POST -G -d @/dev/fd/63 http://10.0.0.0:8080/reload-configuration-as-code
```

I know it's unnecessary, but I had to do a LOT of googling to figure out that trickery. I could remove it and leave it to whomever is using to figure it out if you prefer.

Comment (Angelin01) [reply docs/features/configurationReload.md:16]: I just realized I can make the trickery extremely less trickery if one were to simply write the token to a file in the first place. In my head I couldn't write it to disk in the first place. I'll update it, should make it much simpler.

**Comment by jetersen** [comment]
@Angelin01 no need to squash your commits. We use squash and merge :)

**Review by timja — APPROVED** [review]
Thanks!

**Comment by Angelin01** [comment]
[CI run #1350](https://github.com/jenkinsci/configuration-as-code-plugin/actions/runs/1751242607), which was before the doc changes, appeared to run correctly, but this current one appears stuck, it's been running for nearly 2.5 hours now, don't know what's up with that. Anything else you guys need from me before merging? Otherwise I'll leave it up to you.

**Comment by timja** [comment]
> [CI run #1350](https://github.com/jenkinsci/configuration-as-code-plugin/actions/runs/1751242607), which was before the doc changes, appears to have finished correctly, but [this current one](https://github.com/jenkinsci/configuration-as-code-plugin/actions/runs/1751338235) appears stuck, it's been running for nearly 2.5 hours now, don't know what's up with that. Anything else you guys need from me before merging? Otherwise I'll leave it up to you.

don't worry about it, the main CI has passed.

Codecov is failing I think because there's no test with an environment variable set but empty string:
https://app.codecov.io/gh/jenkinsci/configuration-as-code-plugin/compare/1841/changes

you could add if you have time otherwise fine.

Just waiting to see if @jetersen wants to do a review before merging

**Comment by timja** [comment]
Cool that fixed codecov

**Review by jetersen — APPROVED** [review]
LGTM

**Comment by jetersen** [comment]
Don't worry too much about CI on GitHub as they are currently rolling out images on Ubuntu: https://github.com/actions/virtual-environments

**Pull request merged**
